app/helpers/automated_tests_client_helper.rb

```diff
@@ -260,9 +258,12 @@ def self.get_test_scripts(assignment, user)
 
   def self.request_a_test_run(host_with_port, grouping_id, current_user, submission_id = nil)
 
-    test_server_user = get_test_server_user
     grouping = Grouping.find(grouping_id)
     assignment = grouping.assignment
+    unless assignment.enable_test

```

Comment:
You're right. My point is that maybe this shouldn't be a database column, but rather a computed value based on the "no tests flag" and the presence of runnable test scripts.